lib/puppet/provider/rabbitmqctl.rb

```diff
@@ -8,6 +8,14 @@ def self.rabbitmq_version
     version[1] if version
   end
 
+  # rabbitmqctl version 3.7.9 introduced --no-table-headers flag
+  # which began causing problems for parsing. This resolves the issue
+  # by automatically adding the required flag in the event the RabbitMQ
+  # version is at or above 3.7.9
+  if Puppet::Util::Package.versioncmp(rabbitmq_version, '3.7.9') >= 0
```

Comment:
I think this is a smart way of doing it overall (tho long term, reworking it to use either the JSON output when working with newer rabbitmq, or hitting the API directly, are probably more future-proof), and I guess if we can figure out how to test this part in isolation, we may not need to add new test coverage for all the individual providers, assuming the rest of the format is the same.